src/engine/v3/V3Interpreter.v3

```diff
@@ -1258,7 +1263,42 @@ component V3Interpreter {
 			var shift = u6.view((idx - half_lanes) << u3.+(log2_size, 3));
 			val = extend(view(high >> shift));
 		}
-		stack_push(S.!(val));
+		return val;
+	}
+	def doShuffle(){
+		var b = Values.v_v128(pop());
+		var a = Values.v_v128(pop());
+		var low = doShuffle0(a, b);
+		var high = doShuffle0(a, b);
+		push(Value.V128(low, high));
+	}
+	def doShuffle0(a: (u64, u64), b: (u64, u64)) -> u64 {
+		var res = 0ul;
+		var f = doExtractLane0(_, _, 0, u8.view<u64>, u8.view<u8>);
```

Comment:
Can we remove this partial application here? It will allocate a closure every time the bytecode is executed.

Reply:
Yes I found this problem yesterday and pushed a change. But I did a force push later by mistake so it was overwritten... Updated.